src/prefect/environments/storage/base.py

```diff
@@ -1,34 +1,99 @@
-"""
-Storage objects are used to store Prefect flows so that they can be moved, shared, and
-executed. This is the base Storage class which requires all subclasses to implement
-a `build` function.
-
-The `build` function is meant to take a Prefect flow and store it
-then return that Storage object which contains information about how and where the flow
-is stored.
-"""
+from abc import ABCMeta, abstractmethod
+from typing import Any, TYPE_CHECKING
 
 import prefect
 
+if TYPE_CHECKING:
+    import prefect.core.flow
+
 
-class Storage:
+class Storage(metaclass=ABCMeta):
     """
-    Base class for Storage objects.
+    Base interface for Storage objects.
     """
 
     def __init__(self) -> None:
         pass
 
-    def build(self, flow: "prefect.Flow") -> "Storage":
+    def get_env_runner(self, flow_location: str) -> Any:
```

Comment:
Could you also add a note that this is a way to control a flow run based on environment variables. Since it just says "for running the flow" maybe it should have something in tune with 

`which accepts a collection of environment variables for running the flow. (e.g. specifying an executor with PREFECT__ENGINE__EXECUTOR__etc...)`

Reply:
*that* accepts ;)